builtin_set.go

```diff
@@ -344,3 +344,90 @@ func (r *Runtime) getSet() *Object {
 	}
 	return ret
 }
+
+// estimateMemUsage helps calculating mem usage for large objects.
+// It will sample the object and use those samples to estimate the
+// mem usage.
+func (so *setObject) estimateMemUsage(ctx *MemUsageContext) (estimate uint64, err error) {
+	var samplesVisited, memUsage uint64
+	totalItems := so.m.size
```

Comment:
[q] do we need to check that `so.m` is not nil?

Reply:
Oh yes and I should add a test for it!